src/idl_parser.cpp

```
@@ -2407,14 +2407,17 @@ CheckedError Parser::ParseEnum(const bool is_union, EnumDef **dest) {
   }
 
   if (dest) *dest = enum_def;
-  types_.Add(current_namespace_->GetFullyQualifiedName(enum_def->name),
-             new Type(BASE_TYPE_UNION, nullptr, enum_def));
+  if (types_.Add(current_namespace_->GetFullyQualifiedName(enum_def->name),
+                 new Type(BASE_TYPE_UNION, nullptr, enum_def)))
+    return Error("datatype already exists: " + enum_def->name);
```

It would be better to return the error message with a fully qualified name.